provider/resource_kpi_base_search.go

```diff
@@ -73,30 +73,82 @@ func kpiBaseSearchBase(clientConfig models.ClientConfig, key string, title strin
 }
 
 func (r *resourceKpiBaseSearch) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
-	if req.Plan.Raw.IsNull() || req.State.Raw.IsNull() {
+	if req.State.Raw.IsNull() {
 		return
 	}
 
 	var diags diag.Diagnostics
 	var state, plan KpiBaseSearchState
 	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
 
+	base := models.NewBase(r.client, "", "", "service")
+	params := models.Parameters{
+		Fields: []string{
+			"_key",
+			"title",
+			/* Optional: Uncomment in case verbose error message required
+			"kpis._key", "kpis.title", "kpis.base_search_id", "kpis.alert_period",
+			"kpis.unit", "kpis.aggregate_statop", "kpis.entity_statop", "kpis.threshold_field",
+			"kpis.entity_breakdown_id_fields", "kpis.is_entity_breakdown",*/
+		},
+		Filter: "",
+	}
+
+	// aborts plan in case filter matches > 0 service objects
+	abortLinkedKpis := func(filter string) {
```

Comment:
plan output in case trying to change kpi_base_search metric title that is linked to 2 services:
```terraform
mlukiianova@C02YP2VRLVCJ-mlukiianova kpi_base_search % terraform apply 
itsi_kpi_base_search.DRM_drmAdapter_operation_duration_metrics_by_host_TEST: Refreshing state... [id=cc6ab662-a7e0-da58-c8fd-595367e9b33f]
itsi_service.DRM_Staging_drmDeviceIdManager_test: Refreshing state... [id=01e92c26-83eb-ee57-e69f-8de2869d6952]

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Warning: Filter that found linked KPIs
│ 
│   with itsi_kpi_base_search.DRM_drmAdapter_operation_duration_metrics_by_host_TEST,
│   on generated.tf line 2, in resource "itsi_kpi_base_search" "DRM_drmAdapter_operation_duration_metrics_by_host_TEST":
│    2: resource "itsi_kpi_base_search" "DRM_drmAdapter_operation_duration_metrics_by_host_TEST" {
│ 
│ {"$and": [{"kpis.base_search_id":"cc6ab662-a7e0-da58-c8fd-595367e9b33f"},
│ {"$or":[{"kpis.base_search_metric":
│ "8f68af9f-0746-90e8-d563-decca2dcafdf"},{"kpis.base_search_metric":
│ "14dbfbc8-f866-b7dc-ee33-a6ebb6721974"}]}]}
╵
╷
│ Error: DRM - drmAdapter operation duration metrics by host[TEST] KPI Base Search is linked to the service
│ 
│   with itsi_kpi_base_search.DRM_drmAdapter_operation_duration_metrics_by_host_TEST,
│   on generated.tf line 2, in resource "itsi_kpi_base_search" "DRM_drmAdapter_operation_duration_metrics_by_host_TEST":
│    2: resource "itsi_kpi_base_search" "DRM_drmAdapter_operation_duration_metrics_by_host_TEST" {
│ 
│ _key=01e92c26-83eb-ee57-e69f-8de2869d6952 title=DRM: Staging
│ drmDeviceIdManager[TEST2]
│ 
╵
╷
│ Error: DRM - drmAdapter operation duration metrics by host[TEST] KPI Base Search is linked to the service
│ 
│   with itsi_kpi_base_search.DRM_drmAdapter_operation_duration_metrics_by_host_TEST,
│   on generated.tf line 2, in resource "itsi_kpi_base_search" "DRM_drmAdapter_operation_duration_metrics_by_host_TEST":
│    2: resource "itsi_kpi_base_search" "DRM_drmAdapter_operation_duration_metrics_by_host_TEST" {
│ 
│ _key=627557db-5ba5-c1b2-a234-465db1ea047c title=DRM: Staging
│ drmDeviceIdManager[TEST]
│ 
```